### DISCUSSION TITLE
static lib

### DISCUSSION THREAD

**Discussion by hassansuhail209** [kuzudb/kuzu#4030] [Q&A]
I am using kuzu in my project and I want to use the kuzu static library instead of using dll , how do i use the static lib.

**Comment by benjaminwinger** [comment]
(also answered in a discussion on discord; I'm also answering it here so that it's visible).

This is missing from our docs (it will be added soon), but in short, the easiest way is to use CMake to build and link the static libraries. 

E.g.
```cmake
cmake_minimum_required(VERSION 3.11)
project(example LANGUAGES CXX)

set(BUILD_SHELL FALSE)
include(FetchContent)
FetchContent_Declare(
    kuzu URL https://github.com/kuzudb/kuzu/archive/refs/tags/v0.5.0.zip
)
FetchContent_MakeAvailable(kuzu)

add_executable(example main.c)
target_link_libraries(example kuzu)
```
Note that for the C API you need both `C` and `CXX` in the project `LANGUAGES`, and it won't currently find `kuzu.h` without manually specifying the include directory until #4034 is merged.

If you're using another build system it's more complicated. You need to define `KUZU_STATIC_DEFINE`, and link against the static kuzu library, as well as `antlr4_cypher`, `antlr4_runtime`, `brotlidec`, `brotlicommon`, `utf8proc`, `re2`, `serd`, `fastpfor`, `miniparquet`, `zstd`, `miniz`, `mbedtls`, `lz4` (all of which can be found in the third_party subdirectory of the CMake build directory. E.g. `build/release/third_party/zstd/libzstd.a`) and whichever standard library you're using.
The list will probably change in future; the closest thing to a canonical list can be found in the link_libraries function in the rust API's [build.rs](https://github.com/kuzudb/kuzu/blob/master/tools/rust_api/build.rs) since the targets used in [CMakelists.txt](https://github.com/kuzudb/kuzu/blob/master/src/CMakeLists.txt) include header-only targets and targets whose names don't correspond to their actual library files.
